### PR TITLE
onedrive: 2.3.10 -> 2.3.11

### PR DESCRIPTION
**Pull request by SRGOM**
Also, changed name to onedrive-abraunegg. Shows it's not official, and also gets tracked in repology for a potential auto-update using @ryantm 's bot.

###### Motivation for this change

* Update

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @ doronbehar

### CODE REVIEW

**Comment by SRGOM** [comment]
I don't know how to make r-ryantm track this for autoupdate.... but I would find out and add it

File: pkgs/applications/networking/sync/onedrive/default.nix [hunk 1]
```diff
@@ -29,7 +29,7 @@ stdenv.mkDerivation rec {
     description = "A complete tool to interact with OneDrive on Linux";
     homepage = "https://github.com/abraunegg/onedrive";
     license = licenses.gpl3;
-    maintainers = with maintainers; [ doronbehar ];
+    maintainers = with maintainers; [ doronbehar srgom];
```
Comment (doronbehar) [inline pkgs/applications/networking/sync/onedrive/default.nix]:
```suggestion
    maintainers = with maintainers; [ doronbehar srgom ];
```

File: pkgs/applications/networking/sync/onedrive/default.nix [hunk 2]
```diff
@@ -7,14 +7,14 @@
 }:
 
 stdenv.mkDerivation rec {
-  pname = "onedrive";
-  version = "2.3.10";
+  pname = "onedrive-abraunegg";
```
Comment (jonringer) [inline pkgs/applications/networking/sync/onedrive/default.nix]: I would revert this change, it's so that the bot can pick up the right package in repology

https://repology.org/projects/?search=onedrive&maintainer=&category=&inrepo=&notinrepo=&repos=&families=&repos_newest=&families_newest=

Comment (doronbehar) [reply pkgs/applications/networking/sync/onedrive/default.nix]: I too tend to agree with this opinion. Nothing makes `@skilion`'s version superior besides the fact it was there from before. OTH, what makes `@abraunegg`'s version superior IMO is the fact it is actually maintained - `@skilion`'s version was last updated in 2018.

@SRGOM Is this crucial for repology's updaters to pick it up from Nixpkgs? Even if the answer is yes, I don't think it would matter that much because I don't think other repos update this package that often - meaning @ryantm's bot wouldn't try to update it.

BTW if you are interested in making packages updates more autonomic and with better connection to upstream, you may be interested in: https://github.com/repology/repology-updater/issues/938

**Review by doronbehar — COMMENTED** [review]
.

**Review by doronbehar — APPROVED** [review]
Thanks for collaborating @SRGOM! BTW TBH, I don't use this package, I packaged it just because I considered using it so feel free not to wait for me the next time :)

**Review by doronbehar — APPROVED** [review]

**Comment by SRGOM** [comment]
I'm helping myself... May I ask what you mean by 'so feel free not to wait for me the next time'?

I intend to try and submit a PR that adds a `programs.onedrive.enable` option to create a user service.

**Comment by doronbehar** [comment]
'so feel free not to wait for me the next time' = Feel free to consider yourself as the "main" maintainer.

**Comment by SRGOM** [comment]
Alright, I guess I'll figure out how to add this to ryantm's auto updater

**Comment by jonringer** [comment]
ryantm's auto updater will just scrap information from nix evaluations and try to bump the version and sha256 entries. Just need  write it in a way that isn't specific to the version (E.g. some paths may have version information which would be invalid when updating.)

Essentially ryantm updates are free if you don't make assumptions about being a particular version.

File: pkgs/applications/networking/sync/onedrive/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/applications/networking/sync/onedrive/default.nix]: I'm assuming it goes both ways, there's only 1 package from AUR which has onedrive-abraunegg, while we can get consensus from 8 other package managers by keeping it `onedrive`

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
executable shows usage

```
[2 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/73360
1 package were build:
onedrive
```

**Comment by jonringer** [comment]
oh crap, forgot to look at commits... they should have been squashed

**Comment by SRGOM** [comment]
I had to use the web interface to make these commits. 
I'm going to sound like a luddite but how does one even make github commits with 2FA on?

**Comment by fadenb** [comment]
> but how does one even make github commits with 2FA on? I used to just enter username and password earlier.

Use SSH key (no MFA), a personal access token (no MFA) or a client that supports their web-based login (I believe GiT for Windows does this)

**Comment by SRGOM** [comment]
personal token worked, thank you!
